### ISSUE TITLE
Euclidean

### ISSUE THREAD

**Issue by zenorogue** [ColinEspinas/HumbugEngine#1]
This engine does not appear to be actually capable of doing non-Euclidean geometry.

I see you are doing translations with `x+y`, computing distances between points using (x-y).MagSq(), etc. These are Euclidean formulas. Non-Euclidean geometry uses other formulas.

Portals change the *topology* of the space, but the *geometry* is still Euclidean.

**Comment by ColinEspinas** [comment]
Yes, you are correct, we use tricks to get a non-euclidean feeling (hence the name "humbug").

This was a student project about discovering C++ and OpenGL and is no longer maintained for now.

**Comment by zenorogue** [comment]
Well, you don't get non-Euclidean feeling. There are rectangles everywhere. In non-Euclidean geometry there should be no rectangles (the existence of rectangles, which are not curved and have all edges straight, means that the geometry satisfies Euclid's parallel axiom and thus we are in an Euclidean manifold).
